src/modelconfig.cpp

```diff
@@ -614,12 +614,12 @@ Status ModelConfig::parseNode(const rapidjson::Value& v) {
 
     // Model Cache options
     if (anyShapeSetToAuto())
-        this->setDisableCaching(true);
+        this->setModelCacheState(ModelCacheState::FORCE_CACHE_OFF);
```

Comment:
As stated in documentation: https://github.com/openvinotoolkit/model_server/blob/develop/docs/model_cache.md#enabling-the-cache-functionality
there are 2 scenarios where we disable cache by default:
- shape/batching auto
- w/ custom loader

There is only one scenario wehere we forcibly turn it off, without posibility to turn it on anyway. Doing so should throw an error. And this is w/ custom loader scenario.

This is why I'm worried of using FORCE_CACHE_OFF enum here. My suggestion is:
```
FORCE_CACHE_OFF - when user enforced cache but using custom loader
CACHE_OFF - when cache is disabled (default value)
CACHE_ON - when cache is enabled and not using custom loader
```